src/slice-assign.h

```diff
@@ -1,24 +1,48 @@
 #ifndef VCTRS_SLICE_ASSIGN_H
 #define VCTRS_SLICE_ASSIGN_H
 
+enum vctrs_ownership {
+  vctrs_ownership_owned,
+  vctrs_ownership_shared,
+  vctrs_ownership_unknown
+};
+
+static inline enum vctrs_ownership parse_ownership(SEXP ownership) {
```

Comment:
Move to .c? A header doesn't seem like a good place for this helper, and it doesn't seem important to have it inlinable.